wgpu-types/src/lib.rs

```diff
@@ -2109,6 +2037,467 @@ pub enum TextureFormat {
+                    "etc2-rgb8unorm-srgb" => TextureFormat::Etc2Rgb8UnormSrgb,
+                    "etc2-rgb8a1unorm" => TextureFormat::Etc2Rgb8A1Unorm,
+                    "etc2-rgb8a1unorm-srgb" => TextureFormat::Etc2Rgb8A1UnormSrgb,
+                    "etc2-rgba8unorm" => TextureFormat::Etc2Rgba8Unorm,
+                    "etc2-rgba8unorm-srgb" => TextureFormat::Etc2Rgba8UnormSrgb,
+                    "eac-r11unorm" => TextureFormat::EacR11Unorm,
+                    "eac-r11snorm" => TextureFormat::EacR11Snorm,
+                    "eac-rg11unorm" => TextureFormat::EacRg11Unorm,
+                    "eac-rg11snorm" => TextureFormat::EacRg11Snorm,
+                    other => {
+                        if other.starts_with("astc") {
+                            let mut parts = other.split('-');
+                            parts.next(); // discard "astc"
+                            let block = parts
+                                .next()
+                                .ok_or_else(|| E::invalid_value(Unexpected::Str(s), &self))?;
+                            let channel = parts.collect::<Vec<&str>>().join("-");
+
+                            let block = match block {
+                                "4x4" => AstcBlock::B4x4,
+                                "5x4" => AstcBlock::B5x4,
+                                "5x5" => AstcBlock::B5x5,
+                                "6x5" => AstcBlock::B6x5,
+                                "6x6" => AstcBlock::B6x6,
+                                "8x5" => AstcBlock::B8x5,
+                                "8x6" => AstcBlock::B8x6,
+                                "8x8" => AstcBlock::B8x8,
+                                "10x5" => AstcBlock::B10x5,
+                                "10x6" => AstcBlock::B10x6,
+                                "10x8" => AstcBlock::B10x8,
+                                "10x10" => AstcBlock::B10x10,
+                                "12x10" => AstcBlock::B12x10,
+                                "12x12" => AstcBlock::B12x12,
+                                _ => return Err(E::invalid_value(Unexpected::Str(s), &self)),
+                            };
+
+                            let channel = match channel.as_ref() {
+                                "unorm" => AstcChannel::Unorm,
+                                "unorm-srgb" => AstcChannel::UnormSrgb,
+                                "hdr" => AstcChannel::Hdr,
+                                _ => return Err(E::invalid_value(Unexpected::Str(s), &self)),
+                            };
+
+                            TextureFormat::Astc { block, channel }
+                        } else {
+                            return Err(E::invalid_value(Unexpected::Str(s), &self));
+                        }
+                    }
+                };
+
+                Ok(format)
+            }
+        }
+
+        deserializer.deserialize_any(TextureFormatVisitor)
```

Comment:
```suggestion
        deserializer.deserialize_str(TextureFormatVisitor)
```

I would rather not use `deserialize_any` here, as it is only supported by serializers for self-describing formats. The only implemented `visit_*` implemented is for `str` anyway, so there is not downsides doing that